VSDiagnostics/VSDiagnostics/VSDiagnostics/Utilities/Extensions.cs

```diff
@@ -144,5 +145,37 @@ public static bool IsAsync(this IMethodSymbol methodSymbol)
                 || methodSymbol.ReturnType.MetadataName == typeof(Task).Name
                 || methodSymbol.ReturnType.MetadataName == typeof(Task<>).Name;
         }
+
+        public static bool IsDefinedInAncestor(this IMethodSymbol methodSymbol)
+        {
+            var containingType = methodSymbol?.ContainingType;
+            if (containingType == null)
+            {
+                return false;
+            }
+
+            var interfaces = containingType.AllInterfaces;
+            foreach (var @interface in interfaces)
+            {
+                var interfaceMethods = @interface.GetMembers().Select(containingType.FindImplementationForInterfaceMember);
+                if (interfaceMethods.Any(method => method.Equals(methodSymbol)))
+                {
+                    return true;
+                }
+            }
+
+            var baseType = containingType.BaseType;
+            while (baseType != null)
```

Comment:
Did you really intent the two classes to be disconnected in your example? Or did you mean something like this:

    public class BaseClass
    {
        public virtual void BaseMethod() { }
    }

    public class SubClass : BaseClass
    {
        public void AnotherMethod() { }
    }

Though, regardless what I do, in both examples the loop ends:

![image](https://cloud.githubusercontent.com/assets/460904/11769658/02634502-a1ed-11e5-87ba-4b47e5d1e95f.png)


Shall, I add a commit with the desired unit test as reference?